### PR TITLE
feat(nim): added queries and tests

### PR DESCRIPTION
**Pull request by aMOPel**

### CODE REVIEW

**Comment by HiPhish** [comment]
How and from there did you install the Nim parser? When I execute `TSInstall nim` I get `Parser not available for language "nim"`.

**Comment by aMOPel** [comment]
The pr to add nim to nvim-treesitter was only merged 2 weeks ago. Maybe you need to update you plugin?

File: queries/nim/rainbow-delimiters.scm [hunk 1]
```diff
@@ -0,0 +1,19 @@
+(array_construction "[" @delimiter "]" @delimiter @sentinel) @container
+(tuple_construction "(" @delimiter ")" @delimiter @sentinel) @container
+(tuple_deconstruct_declaration "(" @delimiter ")" @delimiter @sentinel) @container
+(curly_construction "{" @delimiter ":"? @intermediate "}" @delimiter @sentinel) @container
```
Comment (Danielkonge) [inline queries/nim/rainbow-delimiters.scm:4]:
```suggestion
(curly_construction "{" @delimiter ":"? @delimiter "}" @delimiter @sentinel) @container
```

**Review by Danielkonge — COMMENTED** [review]
You should be able to use `@delimiter` for everything being highlighted; `@intermediate` will not work.

**Comment by HiPhish** [comment]
I have made a couple of changes:

- Formatted the patterns to be more like other queries
- Renamed the test file to be more like the other test files
- Fixed the `cast` pattern by merging both patterns into one

We are no longer bound to the limitation of just one opening and one closing node per delimiter. Both the `[`, `]` and the `(`, `)` can be part of the same container on the same level.

```
(cast
  type: "[" @delimiter
  type: "]" @delimiter
  value: "(" @delimiter
  value: ")" @delimiter @sentinel) @container
```

If you agree with my changes please squash the commits into one, then I can merge your PR.

**Comment by aMOPel** [comment]
Thank you :)

All done.

**Comment by HiPhish** [comment]
Merged, thank you very much.

**Pull request merged**